File: php/serialize.php
#!/usr/bin/env php
<?php
/* Serializes object to STDOUT or file.
   The class definition is read from serializeobject.php.

   SPDX-License-Identifier: GPL-3.0-or-later

   This file is part of the deserialization demos:
   https://github.com/PeterMosmans/deserialization-demos
 */

include "serializeobject.php";

$object = new SerializeObject();
$serialized = serialize($object);

if ($argc < 2) {
    /* Write to STDOUT */
    print $serialized;
} else {
    /* Write to file */
    $filename = $argv[1];
    file_put_contents($filename, $serialized);
    print "Object serialized and saved to " . $filename . "\n";
}


?>

File: java/Serialize.java
/* Serializes object to STDOUT or file.
   The class definition is read from MyObject.

   SPDX-License-Identifier: GPL-3.0-or-later

   This file is part of the deserialization demos:
   https://github.com/PeterMosmans/deserialization-demos
 */

package net.gofwd;

import java.io.*;

public class Serialize {

  // Serialize an object to a file
  public static void main(String[] args) {
    MyObject object = new MyObject();
    try {
      if (args.length == 0) {
        ObjectOutputStream out = new ObjectOutputStream(System.out);
        out.writeObject(object);
        out.close();
      } else {
        String filename = args[0];
        FileOutputStream file = new FileOutputStream(filename);
        ObjectOutputStream out = new ObjectOutputStream(file);
        out.writeObject(object);
        out.close();
        file.close();
        System.out.println("Object serialized and saved to " + filename);
      }
    } catch (IOException i) {
      System.out.println("Could not write: " + i);
    }
  }
}
